bids/layout/tests/test_layout.py

```diff
@@ -308,7 +313,11 @@ def test_get_val_enum_any(layout_7t_trt):
                                    suffix='bold', acquisition="*")
     assert len(bold_files) == 2
 
-
+# The following test should have argubly been a failure in pybids
+# but it was not. Session does not exist in dataset, yet since Query
+# Enum is set to optional the query passes even with 'ignore_filters'
```

Comment:
Session is a legitimate entity, so we should be able to search for files where session may be any value or absent. The idea is to be able to write queries that will work regardless of dataset. If we can't query for legit entities that are absent, that's a problem.

Reply:
Right, but shouldn't that behavior depends on `ignore_filters`? Seems kinda weird to me to say `ignore_filters=error` and then not error on `session`.

Reply:
Well an invalid filter is one that doesn't match any files. `session=OPTIONAL` matches all files.

Reply:
Allright, fair enough, I don't feel strongly -- but is this in the actual BIDS spec or just a pyBIDS thing?

Reply:
It is just a pybids thing, but we definitely use it when writing queries. It's useful to say "present but any" (REQUIRED), "absent" (NONE) or "either" (OPTIONAL), and not have to put in a bunch of dataset-dependent logic before constructing your query, or implement your own OPTIONAL and remove it from kwargs.

Reply:
And it's not sufficient to just say `ignore_filters='ignore'` for those types of queries? Is that too "unsafe"?

Reply:
How would you implement it with `invalid_filters='ignore'`? Wouldn't that silently return no files?

Reply:
The way we implement it right now is that we just remove `OPTIONAL` filters from the list. We could check whether the entity is valid and do the same. I think it would also be reasonable for `session=NONE` to return `True` for all files in a dataset without sessions.

Reply:
Yes, it would, but isn't that obvious when writing the query?

I would be in favor of not throwing errors for all BIDS valid core entities by default. Either way, I think this is all open for revision in the new API